core/node/libp2p/rcmgr.go

```diff
@@ -598,3 +602,43 @@ func NetResetLimit(mgr network.ResourceManager, repo repo.Repo, scope string) (r
 
 	return result, nil
 }
+
+func ensureConnMgrMakeSenseVsRessourcesMgr(rcm rcmgr.LimitConfig, cmgr config.ConnMgr) error {
+	if cmgr.Type.WithDefault(config.DefaultConnMgrType) == "none" {
+		return nil // none connmgr, no checks to do
+	}
+	highWater := cmgr.HighWater.WithDefault(config.DefaultConnMgrHighWater)
+	if int64(rcm.System.ConnsInbound) <= highWater {
```

Comment:
This is not compatible with the "just set outbound to infinite and let's trust our own code" doctrine, because `ConnsOutound` will be a really big number.

Now that I think about it, checking `Conns` has the same issue.
I guess we should check:
```go
if min(int64(rcm.System.ConnsInbound), int64(rcm.System.Conns)) <= highWater {
```

Reply:
Actually `rcm.System.ConnsInbound <= rcm.System.Conns` also does the trick and allows for the correct error message.